src/test/java/com/endava/internship/mocking/service/repository/InMemPaymentRepositoryTest.java

```diff
@@ -0,0 +1,100 @@
+package com.endava.internship.mocking.service.repository;
+
+import com.endava.internship.mocking.model.Payment;
+import com.endava.internship.mocking.repository.InMemPaymentRepository;
+import com.endava.internship.mocking.repository.PaymentRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.UUID;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class InMemPaymentRepositoryTest {
+    private PaymentRepository paymentRepository;
+    private Payment payment1;
+    private Payment payment2;
+    private Payment payment3;
+
+
+    @BeforeEach
+    void setUp() {
+        paymentRepository = new InMemPaymentRepository();
+        payment1 = new Payment(1, 100.0, "some string.");
+        payment2 = new Payment(2, 500.0, "some string.");
+        payment3 = new Payment(3, 20000.0, "some string.");
+    }
+
+    @Test
+    void findByIdTestNull() {
+        assertThrows(
+                IllegalArgumentException.class,
+                () -> { paymentRepository.findById(null);},
+                "Payment id is not exists");
```

Comment:
Are you sure that you're checking the message of exception which was thrown? I don't think so. You can use catchThrowable() for retrieving throwable, which, will contain exactly an exception with a message that you can check.